### PR TITLE
Removing duplicate system/syslog ingest pipeline

### PR DESCRIPTION
**Pull request by ycombinator**
## What does this PR do?

This PR removes the duplicate ingest pipeline in the `system/syslog` data stream.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.

## How to test this PR locally

See testing instructions in https://github.com/elastic/kibana/pull/91778.

### CODE REVIEW

**Comment by ycombinator** [comment]
Hi @jen-huang, thanks for finding and reporting the duplicate pipeline. What would be the easiest way to test this fix?

**Review by jen-huang — APPROVED** [review]
Thanks for the quick fix! I have a PR that adds some additional guards on the Kibana side https://github.com/elastic/kibana/pull/91778. The 404 errors weren't breaking either way. I think this is a pretty safe change so I'm happy to test it once this gets deployed out somewhere. Otherwise, there are some testing instructions in my PR.

**Comment by ycombinator** [comment]
> I think this is a pretty safe change so I'm happy to test it once this gets deployed out somewhere. 

Thanks @jen-huang. I'm good with testing this change once this package is published to the snapshot registry then. WDYT @fearful-symmetry?

**Comment by fearful-symmetry** [comment]
@ycombinator I don't know enough how how ingest pipelines are installed to say for sure, but I'm overall fine with it. I'm mostly curious how this has flown under the radar. I don't think we've touched `syslog` for a while?

**Comment by ycombinator** [comment]
>  I'm mostly curious how this has flown under the radar.

Both pipelines for the `system/syslog` dataset are very similar (the only material difference I could find was that the `.yml` pipeline had an extra processor at the end to set `event.type` which was absent in the `.json` pipeline). So my guess is that the net effects on the indexed data were the same (or close enough) that no one noticed any issues. 🤷 

I think what might help is adding a [pipeline test](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) for this data stream. Looks like we have some test files that could be ported over from the [corresponding Filebeat fileset](https://github.com/elastic/beats/tree/master/filebeat/module/system/syslog/test)?

@jen-huang what happens in Kibana if it detects a `default.yml` and `default.json`? I assume it installs both pipelines but they end up having the same names so the first one gets overwritten by the second one? Is there any determinism here, e.g. the `.json` would always be installed before the `.yml` one or vice versa?

**Comment by ycombinator** [comment]
> I think what might help is adding [pipeline tests](https://github.com/elastic/elastic-package/blob/master/docs/howto/pipeline_testing.md) for this data stream. Looks like we have some test files that could be ported over from the [corresponding Filebeat fileset](https://github.com/elastic/beats/tree/master/filebeat/module/system/syslog/test).

Just to clarify, I'm adding these to this PR.

**Comment by jen-huang** [comment]
> @jen-huang what happens in Kibana if it detects a default.yml and default.json? I assume it installs both pipelines but they end up having the same names so the first one gets overwritten by the second one? Is there any determinism here, e.g. the .json would always be installed before the .yml one or vice versa?

Yep, both get installed and the second overwrites the first one. We don't do any sort of logic around the order, just whatever the order happens to be in the package manifest. Looking at [system-0.10.9](https://epr-staging.elastic.co/package/system/0.10.9/), the .json always seems to appear before .yml, so the .yml version would be the one that is "actually" installed.

**Comment by ycombinator** [comment]
@fearful-symmetry @mtojek do you have any tips on how to debug the tests failures in this PR: https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations/detail/PR-725/8/tests/? Initially I'm focussing on getting the pipeline test for `test-darwin-syslog-sample.log` to pass. I assume the other test cases have similar issues.

**Comment by mtojek** [comment]
I think you need to write the config to mark new lines for pipeline tests and probably timezone (as the pipeline doesn't accept lack of it). Something like this:

```
{
  "multiline": {
    "first_line_pattern": "^Dec 13 "
  },
  "fields": {
    "event.timezone": "GMT+2"
  }
}
```

**Comment by ycombinator** [comment]
Thanks for the tip, @mtojek. I added the `multiline` config but looks like I needed the `fields` config too for the time zone field. Trying that now...

**Comment by ycombinator** [comment]
@mtojek I finally gave up and ran `elastic-package test pipeline --data-stream syslog --generate`. After doing this and making a couple of other tweaks to the test case config files, all pipeline tests for the `system/syslog` data stream are now passing.

Take a look at the `test-darwin-syslog-sample.log-expected.json` file and compare it with the `test-darwin-syslog-sample.log` file. The expected file has only one event while the corresponding raw log file has three events. This looks wrong 🙂 and the tests should not be passing, but they are! I think maybe there is a bug in the pipeline test runner?

**Comment by ycombinator** [comment]
@mtojek ignore my previous comment 😄. I had to tweak the `multiline.first_line_pattern` in the test case config. 🤦

**Comment by ycombinator** [comment]
@fearful-symmetry I added test case files for the `system/syslog` data stream. Would you mind re-reviewing this PR? Thanks!

**Review by jen-huang — APPROVED** [review]

File: packages/system/changelog.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
 # newer versions go on top
+- version: "0.11.1" # unreleased
```
Comment (mtojek) [inline packages/system/changelog.yml:2]: I'm pretty sure one day somebody will forget about this comment. I think we need a validator which check consistency between changelog and manifest's version.

**Review by mtojek — APPROVED** [review]

File: packages/system/changelog.yml (hunk 1 above)
Comment (ycombinator) [reply packages/system/changelog.yml:2]: Agreed. There is also a question about what the unreleased version should be. I'll start a discussion in the package spec repo.

Comment (ycombinator) [reply packages/system/changelog.yml:2]: https://github.com/elastic/package-spec/issues/140

**Pull request merged**